src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1143,7 +1143,16 @@ ReturnCode_t DataWriterImpl::set_qos(
     {
         return ReturnCode_t::RETCODE_IMMUTABLE_POLICY;
     }
-    set_qos(qos_, qos_to_set, !enabled);
+    set_qos(qos_, qos_to_set, enabled);
+    if (enabled && qos_.reliability().kind == eprosima::fastrtps::RELIABLE_RELIABILITY_QOS)
+    {
+        // Update times and positive_acks attributes on RTPS Layer
+        WriterAttributes w_att;
+        w_att.times = qos_.reliable_writer_qos().times;
+        w_att.disable_positive_acks = qos_.reliable_writer_qos().disable_positive_acks.enabled;
+        w_att.keep_duration = qos_.reliable_writer_qos().disable_positive_acks.duration;
+        writer_->updateAttributes(w_att);
+    }
```

Comment:
What do you think on doing something [this](https://github.com/eProsima/Fast-DDS/files/12747974/patch.txt) in here ? 
Perhaps, instead of always updating the `reliable_writer_qos` if we are enabled and we are reliable, we could check if that `qos` changed since last time, store it in a `should_update` flag and then checking if the flag was up. It is just a little optimization, a little NIT.